app/serializers/site_serializer.rb

```diff
@@ -246,6 +247,10 @@ def include_anonymous_default_sidebar_tags?
     scope.anonymous? && !SiteSetting.legacy_navigation_menu? && SiteSetting.tagging_enabled && SiteSetting.default_sidebar_tags.present?
   end
 
+  def whispers_allowed_groups_names
+    SiteSetting.whispers_allowed_groups_map&.map { |id| Group.where(id: id).pluck_first(:name) } if scope.can_see_whispers?
```

Comment:
Super small thing, but in the serializers we can use `include_METHOD_NAME?` to determine whether there should be a property with that METHOD_NAME in the serializer. So, here, we can do: 

```
def include_ whispers_allowed_groups_names?
  scope.can_see_whispers?
end
```

And that will do the same thing as `if scope.can_see_whispers?`, except that when false, `whispers_allowed_groups_names` will not be included at all in the serializer. 

This is a tiny bit cleaner, no need to add `whispers_allowed_groups_names: nil` when it's nil. (This also means that the site_response.json format only allows an array, and not nil.)

Reply:
Oh, I didn't know about that functionality. It's definitely cleaner. Thanks for pointing that out!